plugins/orchestrator-backend/src/service/api/v1.ts

```diff
@@ -146,21 +150,31 @@ export namespace V1 {
     businessKey: string | undefined,
   ): Promise<WorkflowExecutionResponse> {
     const definition = await dataIndexService.getWorkflowDefinition(workflowId);
-    const serviceUrl = definition.serviceUrl;
-    if (!serviceUrl) {
+    if (!definition) {
+      throw new Error(`Couldn't fetch workflow definition for ${workflowId}`);
+    }
+    if (!definition.serviceUrl) {
       throw new Error(`ServiceURL is not defined for workflow ${workflowId}`);
     }
     const executionResponse = await sonataFlowService.executeWorkflow({
       workflowId,
       inputData: reqBody,
-      endpoint: serviceUrl,
+      endpoint: definition.serviceUrl,
       businessKey,
     });
 
     if (!executionResponse) {
       throw new Error(`Couldn't execute workflow ${workflowId}`);
     }
 
+    // Making sure the instance data is available before returning
+    await retryAsyncFunction({
```

Comment:
should we use ` retryAsyncFunction` for all the endpoints that interacts with dataindex? e.g. abort workflow, get instance by id, get workflows list, etc.)

Reply:
I'd use it only by demand. But let's see first if this code resolves the issue we are seeing.